### PR TITLE
Update to Play 2.5

### PR DESCRIPTION
**Pull request by adamnfish**
Play application state (WS / config) are now injected into the Controller.

This is to update to the latest version of Play before I start adding lifecycle-based components to the application (which would make it much harder to migrate Play version).

### CODE REVIEW

File: app/controllers/Healthcheck.scala [hunk 1]
```diff
@@ -1,8 +1,10 @@
 package controllers
 
+import javax.inject.Inject
+
 import play.api.mvc._
 
-class Healthcheck extends Controller {
+class Healthcheck @Inject() extends Controller {
```
Comment (sihil) [inline app/controllers/Healthcheck.scala]: Ugg. So damn unpretty.

**Comment by sihil** [comment]
:+1: Also - shall we fork Play already and lose the DI?

File: app/controllers/Healthcheck.scala (hunk 1 above)
Comment (adamnfish) [inline app/controllers/Healthcheck.scala]: I know!!!
To be fair, the other controller comes out looking a bit neater, although at the cost of it using impenetrable Dark Magiks.

File: app/controllers/AMIable.scala [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-import prism.Prism
-import prism.PrismLogic
+import prism.{Prism, PrismLogic}
 
-import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.ExecutionContext
 
 
-class AMIable extends Controller {
+class AMIable @Inject()(ws: WSClient, playConfig: Configuration)(implicit exec: ExecutionContext) extends Controller {
```
Comment (cb372) [inline app/controllers/AMIable.scala]: :mask:

Comment (adamnfish) [inline app/controllers/AMIable.scala]: I may switch to the compiled version at some point, but I just wanted to upgrade to Play 2.5 with as small a change as possible. Sorry.
As we've discussed before though, I'm deeply unsatisfied with both approaches so I'm still on the fence about moving off the "simpler" (sic) and better documented default version.

**Pull request merged**